File: lib/wasm-micro-runtime-fast-jit-06-29-2022/assembly-script/samples/event_subscriber.ts
/*
 * Copyright (C) 2019 Intel Corporation.  All rights reserved.
 * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 */

// The entry file of your WebAssembly module.
import * as console from "../wamr_app_lib/console"
import * as timer from "../wamr_app_lib/timer"
import * as request from "../wamr_app_lib/request"

export function on_init() : void {
    request.subscribe_event("alert/overheat", (req) => {
        console.log("### user over heat event handler called:");

        console.log("");
        console.log("    " + String.UTF8.decode(req.payload) + "\n");
    })
}

export function on_destroy() : void {

}


/* Function below are requred by wamr runtime, don't remove or modify them */
export function _on_timer_callback(on_timer_id: i32): void {
    timer.on_timer_callback(on_timer_id);
}

export function _on_request(buffer_offset: i32, size: i32): void {
    request.on_request(buffer_offset, size);
}

export function _on_response(buffer_offset : i32, size: i32): void {
    request.on_response(buffer_offset, size);
}

File: lib/wasm-micro-runtime-fast-jit-06-29-2022/samples/simple/wasm-apps/event_subscriber.c
/*
 * Copyright (C) 2019 Intel Corporation.  All rights reserved.
 * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 */

#include "wasm_app.h"
#include "wa-inc/request.h"

void
over_heat_event_handler(request_t *request)
{
    printf("### user over heat event handler called\n");

    if (request->payload != NULL && request->fmt == FMT_ATTR_CONTAINER)
        attr_container_dump((attr_container_t *)request->payload);
}

void
on_init()
{
    api_subscribe_event("alert/overheat", over_heat_event_handler);
}

void
on_destroy()
{
    /* real destroy work including killing timer and closing sensor is
       accomplished in wasm app library version of on_destroy() */
}
